test/core/end2end/generate_tests.bzl

```diff
@@ -468,33 +480,68 @@ def grpc_end2end_tests():
             #print(_compatible(fopt, topt), f, t, fopt, topt)
             if not _compatible(fopt, topt):
                 continue
-
             test_short_name = str(t) if not topt.short_name else topt.short_name
-            native.sh_test(
-                name = "%s_test@%s" % (f, test_short_name),
-                data = [":%s_test" % f],
-                srcs = ["end2end_test.sh"],
-                args = [
-                    "$(location %s_test)" % f,
-                    t,
-                ],
-                tags = ["no_linux"] + _platform_support_tags(fopt),
-                flaky = t in fopt.flaky_tests,
-            )
 
+            # Non-Linux platforms
```

Comment:
Can't we just replace all of this with the following:

```
grpc_cc_test(
    name = "%s_test@%s" % (f, test_short_name),
    data = [":%s_test" % f],
    args = [t],
    tags = _platform_support_tags(fopt),
    flaky = t in fopt.flaky_tests,
    uses_polling = True,
)
```

That way, the `grpc_cc_test()` macro will automatically expand the EE and polling engine parameters for us, and we don't have to duplicate the code here.

Similar thing for the nosec tests below.